### DISCUSSION TITLE
How can I remove the background color of the signature helper?

### DISCUSSION THREAD

**Discussion by mapringg** [LazyVim/LazyVim#419] [Q&A] [answered]
I would like to keep the bold text, but get rid of the background.

<img width="554" alt="Screenshot 2566-03-13 at 02 21 13" src="https://user-images.githubusercontent.com/15067237/224567964-e860b879-b228-4cdd-a344-8315d92cd0a5.png">

**Comment by folke** [comment, marked as answer]
The hl group you're looking for is NoicePopup. Alternativaly you can set a different winhighlight for the noice `hover` view. Check the noice docs

**Reply by mapringg** [reply]
I think I am close to it, but I got stuck after getting the border. The selected item still has a different background. This is my current config:

```
return {
  "folke/noice.nvim",
  opts = {
    presets = {
      lsp_doc_border = true,
    },
    views = {
      hover = {
        win_options = {
          winhighlight = {
            Normal = "NoicePopup",
            FloatBorder = "NoicePopup",
            CursorLine = "NoicePopup",
            PmenuMatch = "NoicePopup",
          },
        },
      },
    },
  },
}
```

Edit: The config still does not work, but I believe I am now close to the solution.

<img width="744" alt="Screenshot 2566-03-13 at 03 08 59" src="https://user-images.githubusercontent.com/15067237/224570650-1d477091-e96b-440c-a884-a39c2323a604.png">

**Reply by folke** [reply]
Your colorscheme probably sets a bg for NoicePopup, so you need to clear that yourself, or just set another hl group in the winhighlights

**Reply by mapringg** [reply]
I was not able to change the active parameter background. End up having to set this, so everything inside is the same color:

```
views = {
  hover = {
    win_options = {
      winhighlight = {
        Normal = "LspSignatureActiveParameter",
        FloatBorder = "LspSignatureActiveParameter",
      },
    },
  },
}
```

<img width="890" alt="Screenshot 2566-03-13 at 10 57 11" src="https://user-images.githubusercontent.com/15067237/224604955-0573a5a1-eb99-40e3-8ae5-427aea393b87.png">

Edit: Is it possible to disable noice and get the documentation and hover to look like nvim-cmp?

<img width="1183" alt="Screenshot 2566-03-13 at 11 21 39" src="https://user-images.githubusercontent.com/15067237/224608005-7f84e1b9-eb64-4ba6-bdee-c9988f9895c5.png">

**Reply by mapringg** [reply]
I got it working now.

Autocomplete:
![Screenshot 2566-03-13 at 15 33 24](https://user-images.githubusercontent.com/15067237/224648047-3fac73f6-982a-46d3-b1e1-d23e4cbe7153.png)

Hover:
![Screenshot 2566-03-13 at 15 25 26](https://user-images.githubusercontent.com/15067237/224646326-4444380b-f223-4b97-93a0-c82b9c776655.png)

Signature:
![Screenshot 2566-03-13 at 15 25 35](https://user-images.githubusercontent.com/15067237/224646359-870c5144-dc32-4eee-a87b-81cc67acf6ff.png)

**Reply by danielo515** [reply]
What were you tring to do? I want to add borders to the CMP popus, How you made it?

**Reply by danielo515** [reply]
Ok, this did the trick:

```lua
return {
  "hrsh7th/nvim-cmp",
  opts = function(_, opts)
    local cmp = require("cmp")
    opts.window = {
      completion = cmp.config.window.bordered(),
      documentation = cmp.config.window.bordered(),
    }
  end,
}
```

**Reply by mkhai9x** [reply]
hi hi, I just updated my LazyVim and somehow this no longer works for me, I wonder everyone is the same? 
```
return {
  "hrsh7th/nvim-cmp",
  opts = function(_, opts)
    local cmp = require("cmp")
    opts.window = {
      completion = cmp.config.window.bordered(),
      documentation = cmp.config.window.bordered(),
    }
  end,
}
```

**Reply by thomassun** [reply]
me too, it doesn't work for me.

**Reply by RBird111** [reply]
For anyone who stumbles here looking for a way to add borders to the hover/signature menus, here's how I managed to get it working:
```lua
return {
  "folke/noice.nvim",
  opts = function(_, opts)
    opts.presets.lsp_doc_border = true
  end,
}
```

Relevant docs:
[LazyVim UI](https://www.lazyvim.org/plugins/ui#noicenvim)
[Noice Config](https://github.com/folke/noice.nvim?tab=readme-ov-file#%EF%B8%8F-configuration)
